### PR TITLE
BlockSoftmax: fix guard value for fp16

### PR DESCRIPTION
**Pull request by madamczykhabana**

### CODE REVIEW

**Comment by madamczykhabana** [comment]
gsm_8k results

before:
|     Tasks     |Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|---------------|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k_cot_llama|      3|flexible-extract|     8|exact_match|↑  |0.8362|±  |0.0102|
|               |       |strict-match    |     8|exact_match|↑  |0.8347|±  |0.0102|

after:
|     Tasks     |Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|---------------|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k_cot_llama|      3|flexible-extract|     8|exact_match|↑  |0.8431|±  |  0.01|
|               |       |strict-match    |     8|exact_match|↑  |0.8423|±  |  0.01|

vllm (pretrained=/mnt/weka/data/pytorch/llama3.1/Meta-Llama-3.1-8B-Instruct/,dtype=bfloat16,max_num_seqs=128)

**Review by szutenberg — APPROVED** [review]
